implementations/rust/ockam/ockam_command/src/enroll/oidc_service.rs

```diff
@@ -520,4 +580,66 @@ mod tests {
         assert!(code.is_ok());
         assert_eq!(code.unwrap(), "12345".to_string())
     }
+
+    enum ClientState {
+        FirstCall,
+        SecondCall,
+        Finished
+    }
+
+    struct Client {
+        state: Arc<Mutex<ClientState>>
+    }
+
+    #[async_trait]
+    impl GetUserInfo for Client{
+        async fn get_user_info(&self, _token: &OidcToken) -> Result<UserInfo> {
+
+                let mut guard = self.state.lock().unwrap();
```

Comment:
@talir0drigues you're going to need to format your code with `cargo fmt` to pass the CI checks.